### ISSUE TITLE
Recommended usage for cross-language monorepos

### ISSUE THREAD

**Issue by Avasam** [dprint/dprint#918]
I just saw https://github.com/dprint/dprint/issues/859 and that dprint now supports TOML and YAML out of the box, which is awesome. 
But I'm still not sure how to go about multi-languages monorepos (like a frontend/backend), or even a monorepo that doesn't have a root node_module.

Asking devs to do a global install means I can't help ensure a consistent version is used (dprint changing how it treats its configs in the past has broken across versions, not a bad thing since I could pin in in package.json).
Installing dprint in the frontend repo to pin its version means that when run as an npm script it won't touch root-level and backend config files and markdowns.

At work I'm trying to keep tooling as easy to setup and run as possible, if someone wants to lint their project, they'd only need to run one premade command. So for example right now I'm including dprint as part of devDependencies in pakcage.json, and telling devs to simply add this to their npm scripts so they have 1 command for lint and 1 for fix:
```json
"scripts": {
    "eslint": "eslint ./ --ignore-path .gitignore",
    "lint": "dprint check && npm run eslint",
    "lint:fix": "dprint fmt && npm run eslint -- --fix",
}
```
eg: https://github.com/BesLogic/releaf-canopeum/blob/cbd877ce770803dc8c540d8bd281955c76b346a9/canopeum_frontend/package.json#L10-L12

This works great for js-only projects. Or JS monorepos using npm workspaces.

For backends of a monorepo (let's say Python), we usually just make a single script that runs all the different tools (thanks Ruff for reducing that number, but there's still type-checkers):
Once again, example: https://github.com/BesLogic/releaf-canopeum/blob/cbd877ce770803dc8c540d8bd281955c76b346a9/canopeum_backend/scripts/checkers.py

Whilst dprint is so fast I don't mind running it on extra files I don't need to, it doesn't make sense to run slower tools on irrelevant projects, so I would want to move a "lint everything" command to the root of a mono-repo.

Even with https://github.com/dprint/dprint/issues/859, it wouldn't help devs that work only frontend or only backend (especially for another language than Python, let's say C#, or PHP), as they would have to install another projet/language's libs in the monorepo just to format the project they're working on.
It doesn't help that when the node_module isn't at root, we have to specify it for dev's format-on-save to even work. eg: `"dprint.path": "frontend_project/node_modules/dprint/dprint"` (https://github.com/BesLogic/releaf-canopeum/blob/cbd877ce770803dc8c540d8bd281955c76b346a9/.vscode/settings.json#L94C3-L94C64) (which once again, only works assuming full-stack)

I can have the CI check it independently, but once again that requires managing tools installation differently and that's an area we have to be extra cautious about keeping simple w/o speedbumps. Especially if CI checks fails on formatting config files that the dev can't easily run a "fix all" command locally and has to jump through hoops to get to it.

It doesn't help that CI autofixes through pre-commit doesn't work unless you pay for it (which isn't great for non-commercial, outside of work projects either) https://github.com/dprint/dprint/issues/442#issuecomment-2219067045 (we already have a custom solution for Azure DevOps, I'd have to look into https://github.com/EndBug/add-and-commit or https://github.com/stefanzweifel/git-auto-commit-action .

I thought about changing the "lint" commands to run from the parent folder (which is where the dprint.json is anyway):
```json
"scripts": {
    "eslint": "eslint ./ --ignore-path .gitignore",
    "lint": "dprint check .. && npm run eslint",
    "lint:fix": "dprint fmt  .. && npm run eslint -- --fix",
}
```
So that at least we still have that 1 "format everything" command available without requirering additional setup for frontend devs. But I'm getting the error:
```
No files found to format with the specified plugins at [...]\frontend_project. You may want to try using `dprint output-file-paths` to see which files it's finding or run with `--allow-no-files`.
```
Edit: I can do `cd .. && dprint check && cd frontend_project`, quite verbose, but that's fine for a npm command that is meant to shorten that complexity.

**What I think is currently best, but can we do better ?**
Maybe I should just tell people to install dprint globally. Hopefully dprint prompts to update itself once in a while (like npm and pip do), and we have the burden of keeping up with any new version breakage.
That should avoid having to configure `dprint.path` in VSCode workspace settings. Manual "all-in-one" scripts can call "dprint" w/o caring too much about its location. npm projects usign npm scripts should automatically pickup the "dprint" from node_modules. Cross-language monorepos will unfortunately not be able to pin the version for devs and will require *a single* extra job step to format root-level files (and `npm run lint:fix` as run whitin a sub-project will at least still format that project's files). Could lead to dprint running twice on the CI, but as I said, that's not too bad.

tl;dr: How should I setup dprint for cross-language monorepos so that root-level (and otherwise not inside the frontend/JS project) can be formatted easily through CLI and VSCode's "Format on Save", whilst minimizing the need for extra setup for devs creating new projects, joining existing projects or updating their tooling after comming back to a project.

**Comment by Avasam** [comment]
I've simply been running some variant of "cd .. && dprint && back_to_project && language_specific_linter" for a while now and have been satisfied.

JS mono-repos the tools and configurations live at the root anyway.

C# (Roslyn & CSharpier) and Python (Ruff) backends I don't run through dprint anyway.

Getting into a mixed multi-language monorepo with a nested frontend monorepo soon. A project like that can afford a completely separate run of dprint on the entire thing (or to pass in args/configs to ignore the projects already covered by subcommands) to catch files outside said projects.

**Comment by adrian-gierakowski** [comment]
You can use a nix shell (and optionally direnv) to provide consistent dprint version both locally and in CI

**Comment by Avasam** [comment]
I'm not entirely certain what nix is, but from a (very quick) look at it, I'm not gonna learn then try to enforce on an entire company and a few other cooperative projects an entirely different, non-Windows native development ecosystem just to run a CLI a few folders up.
